.buildkite/pipeline-utils/buildkite/client.ts

```diff
@@ -78,6 +130,8 @@ export class BuildkiteClient {
       },
     });
 
+    this.exec = config.exec ?? execSync;
```

Comment:
Added an option to override exec with a mock, if we want to play around with the scripts locally